File: src/main/resources/output/IFC4X3_python/Entities/IfcRepresentationItem/IfcGeometricRepresentationItem/IfcCurve/IfcPcurve/IfcPcurve.py
class IfcPcurve(IfcCurve):
    def __init__(self):
        self.BasisSurface: IfcSurface = None
        self.ReferenceCurve: IfcCurve = None

    # === WHERE CLAUSES ===
    # DimIs2D : ReferenceCurve.Dim = 2

File: src/main/resources/output/IFC4X3_csharp/Entities/IfcRepresentationItem/IfcGeometricRepresentationItem/IfcCurve/IfcPcurve/IfcPcurve.cs
public class IfcPcurve : IfcCurve
{
    public IfcSurface BasisSurface { get; set; }
    public IfcCurve ReferenceCurve { get; set; }

    // === WHERE CLAUSES ===
    // DimIs2D : ReferenceCurve.Dim = 2
}
